docs/usage-with-routing.stories.mdx

```diff
@@ -121,11 +133,19 @@ const CarbonButtonWithReactRouter = ({ to, children }) => {
 
 const CarbonLinkWithReactRouter = ({ to, children }) => {
   const navigate = useNavigate();
-  const onClick = useCallback(() => {
-    navigate(to);
-  }, [to, navigate]);
+  const onClick = useCallback(
+    (event) => {
+      event.preventDefault();
+      navigate(to);
+    },
+    [to, navigate]
+  );
 
-  return <Link onClick={onClick}>{children}</Link>;
+  return (
+    <Link onClick={onClick} href={to}>
```

Comment:
Same comment as above - now both the href and the `onClick` will be navigating, it should be one or the other I think.

Maybe add a separate example of a Link using href here, that isn't also navigating within the `onClick` function